packages/turbo-repository/rust/src/lib.rs

```diff
@@ -110,26 +144,41 @@ impl Workspace {
     /// Finds and returns a map of packages within the workspace and its
     /// dependents (i.e. the packages that depend on each of those packages).
     #[napi]
-    pub async fn find_packages_and_dependents(
-        &self,
-    ) -> Result<HashMap<String, Vec<String>>, Error> {
+    pub async fn find_packages_with_graph(&self) -> Result<SerializablePackages, Error> {
         let packages = self.find_packages().await?;
 
         let workspace_path = match AbsoluteSystemPath::new(self.absolute_path.as_str()) {
             Ok(path) => path,
-            Err(e) => return Err(Error::from_reason(e.to_string())),
+            Err(e) => {
+                print!("Error: {:?}", e);
+                return Err(Error::from_reason(e.to_string()));
+            }
         };
 
-        let map: HashMap<String, Vec<String>> = packages
+        // Create a map of package names to their dependents and dependencies
+        //  {
+        //    "package-path": {
+        //      "dependents": ["dependent1_path", "dependent2_path"],
+        //      "dependencies": ["dependency1_path", "dependency2_path"]
```

Comment:
Does this include non-workspace dependencies?

Reply:
What do you mean? These will be the list of _packages_ in the workspace, but just their relative paths relative to the root of the workspace

Reply:
Yea that's what I meant.

Reply:
The api kinda misleads to make it seem like you return all dependencies not just the ones in the same workspace. 
But I guess it's just something you read in the docs once.